src/vs/workbench/contrib/files/browser/files.contribution.ts

```diff
@@ -239,7 +239,7 @@ configurationRegistry.registerConfiguration({
 		},
 		'files.watcherExclude': {
 			'type': 'object',
-			'default': isWindows /* https://github.com/microsoft/vscode/issues/23954 */ ? { '**/.git/objects/**': true, '**/.git/subtree-cache/**': true, '**/node_modules/*/**': true, '**/.hg/store/**': true } : { '**/.git/objects/**': true, '**/.git/subtree-cache/**': true, '**/node_modules/**': true, '**/.hg/store/**': true },
+			'default': isWindows /* https://github.com/microsoft/vscode/issues/23954 */ ? { '**/.git/objects/**': true, '**/.git/subtree-cache/**': true, '**/node_modules/*/**': true, '**/.hg/store/**': true } : { '**/.git/objects/**': true, '**/.git/subtree-cache/**': true, '**/node_modules/**/*': true, '**/.hg/store/**': true },
```

Comment:
@Aaaaash any reason we wouldn't just pick the exact same pattern as on Windows? From my testing, the advantage of `'**/node_modules/*/**'` over `'**/node_modules/**/*'` seems to be that you would see changes to top level folders in the `node_modules` folder, but nothing deeper. So essentially the explorer refreshes nicely on operations such as `yarn add <module>`. 

With your suggested pattern, I only see `node_modules` folder appearing and hiding, but not changes inside the folder.

Reply:
Ping @Aaaaash

Reply:
Sorry for the late reply, you are right, so we can use the same pattern on Windows and Linux/macOS ？

Reply:
Yeah I think so, can you update the PR?

Reply:
Updated